ga4gh/datamodel/reads.py

```diff
@@ -325,8 +343,17 @@ def getReadAlignments(self, referenceId=None, start=None, end=None):
         referenceId, start, end = self.sanitizeAlignmentFileFetch(
             referenceId, start, end)
         readAlignments = samFile.fetch(referenceId, start, end)
-        for readAlignment in readAlignments:
-            yield self.convertReadAlignment(readAlignment)
+        if self._filterReads:
+            for readAlignment in readAlignments:
+                for tag in readAlignment.tags:
```

Comment:
It's tempting to just make a dict out of this... This is potentially a performance bottleneck, but it might not matter. If it does, we can always change it back to something like you have now. So, specifically, I'd put:
```
# This may be a performance bottleneck; pysam returns a list of tuples not a dict.
tags = dict(readAlignment.tags)
if 'RG' in tags:
    # etc
```